### DISCUSSION TITLE
RFC tracking discussion: type-system DSL

### DISCUSSION THREAD

**Discussion by vilkinsons** [blockprotocol/blockprotocol#540] [RFC]
@indietyp has opened an early draft RFC proposing a type-system DSL in https://github.com/blockprotocol/blockprotocol/pull/532

**Comment by indietyp** [comment]
I'd like to give the DSL a proper name (which is hard), after some brainstorming I came up with:

* Block Config (`.bc`) (💡 : `HCL`)
* Block Protocol Configuration Language (`.bcl`) (💡 : `HCL`)
* Block Protocol Config (`.bpc`) (💡 : `HCL`)
* Block Protocol Language (`.bpl`) (💡 : `HCL`)
* Block Protocol Object Notation (`.bon`, `.bp` or `.bpon`) (💡 : `JSON`)
* moss (💡: random noun)

Suggestions are very much welcome!

**Reply by vilkinsons** [reply]
I lean towards `.bpl` or `.bpcl`(respectively Block Protocol Language and Block Protocol Configuration Language).

**Reply by ghost** [reply]
Hi nonparibus!

Have you thought about having a database like orbit-db to use in DSL?

**Reply by vilkinsons** [reply]
> Have you thought about having a database like orbit-db to use in DSL?

@codehangen Can you elaborate?

**Reply by ghost** [reply]
Hi nonparibus. 

> Can you elaborate?

I thought there would be a need to have a block protocol database to store certain component settings.

**Reply by vilkinsons** [reply]
@codehangen The Block Protocol is designed to be agnostic to the kind of datastore users bring. We'd like to avoid being prescriptive about this, and I'm not sure the suggestion relates to this thread (around the DSL?)

**Reply by ghost** [reply]
Hi nonparibus.

> @codehangen The Block Protocol is designed to be agnostic to the kind of datastore users bring. 

I understand this point of view that the block protocol was designed to be independent of the type of data storage that users bring. So, reading your comment, I realize that my idea is wrong. 

*thank you for feedback.*
